Content.Shared/Mobs/Events/BeforeThresholdMobStateUpdateEvent.cs

```diff
@@ -0,0 +1,11 @@
+namespace Content.Shared.Mobs.Events;
+
+/// <summary>
+/// Raised before MobThresholdSystem changes MobState and can cancel it.
+/// </summary>
+public sealed class BeforeThresholdMobStateUpdateEvent(EntityUid target, MobState changeMobStateFrom, MobState changeMobStateTo) : CancellableEntityEventArgs
+{
+    public readonly EntityUid Target = target;
+    public readonly MobState ChangeMobStateFrom = changeMobStateFrom;
+    public readonly MobState ChangeMobStateTo = changeMobStateTo;
+}
```

Comment:
This file should be somewhere under the _RMC14 folder in Content.Shared, not in an upstream folder